packages/foam-vscode/src/features/janitor.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  window,
+  workspace,
+  ExtensionContext,
+  commands,
+  Position,
+  Range
+} from "vscode";
+import fs = require("fs");
+import { FoamFeature } from "../types";
+import {
+  applyTextEdit,
+  generateLinkReferences,
+  generateHeading,
+  Foam,
+  Note
+} from "foam-core";
+
+import { bootstrap, getConfig } from "../extension";
+import { includeExtensions } from "../settings";
+
+const feature: FoamFeature = {
+  activate: async (context: ExtensionContext, _foamPromise: Promise<Foam>) => {
+    context.subscriptions.push(
+      commands.registerCommand("foam-vscode.janitor", janitor)
+    );
+  }
+};
+
+async function janitor() {
+  try {
+    const outcome = await runJanitor();
+    if (outcome.processedFileCount === 0) {
+      window.showInformationMessage(
+        "Foam Janitor didn't file any notes to clean up"
+      );
+    } else if (!outcome.changedAnyFiles) {
+      window.showInformationMessage(
+        `Foam Janitor checked ${outcome.processedFileCount} files, and found nothing to clean up!`
+      );
+    } else {
+      window.showInformationMessage(
+        `Foam Janitor checked ${outcome.processedFileCount} files and updated ${outcome.updatedDefinitionListCount} out-of-date definition lists and added ${outcome.updatedHeadingCount} missing headings. Please check the changes before committing them into version control!`
+      );
+    }
+  } catch (e) {
+    window.showErrorMessage(`Foam Janitor attempted to clean your workspace but ran into an error. Please check that we didn't break anything before committing any changes to version control, and pass the following error message to the Foam team on GitHub issues:
+    ${e.message}
+    ${e.stackTrace}`);
+  }
+}
+
+async function runJanitor() {
+  const foam = await bootstrap(getConfig());
```

Comment:
`foam` is passed in the `activate` function of the feature, so that extensions don't need to bootstrap individually. in the `activate` function we should wait for the promise, then pass `foam` as a param in this function

Reply:
@riccardoferretti ha yeah, I noticed this myself and already fixed it sneakily on master:
https://github.com/foambubble/foam/commit/7c6874d52844cffc9936efd056848f7e9dd7ce9c